Add Android 12 support for minimediaservice

NotKit opened the pull request:
Upstream the changes from Halium as there have been Android 12-based adaptations using this for a while.

NotKit committed:
Fix build for droid version 12

FakeAudioPolicyService is currently not implemented as
AudioPolicyService interface has been heavily changed
to use generated AIDL code.

Change-Id: Idaf000ef946ddb602a48ae110b46e3bfcace8d12

NotKit committed:
Add startWatchingModeWithFlags to service_12_0_0.h FakeAppOps

Fixes compilation after 2023-12 ASB changes.

See https://review.lineageos.org/c/LineageOS/android_frameworks_native/+/376795

NotKit committed:
Implement FakeAudioPolicyService for Android 12

mlehtima commented:
One question, does Android 12 also need changes similar to this https://github.com/sailfishos/droidmedia/pull/131. I'm pretty sure it does because of https://github.com/LineageOS/android_frameworks_av/blob/lineage-19.1/media/libstagefright/MediaCodec.cpp#L1824. It's possible Halium does things differently but that seems to be causing issues on Sailfish OS way of porting. Otherwise changes look good.

NotKit commented:
It probably does need it then, I will see if I can get FakePackageNative service to compile. We previously patched out that check (https://github.com/Halium/hybris-patches/commit/f69cce2a3d064c7bf407c321ef537bfe0ae8a5fe) since it was the only place package_native service was used.

NotKit committed:
Implement FakePackageManagerNative for Android 12

Note that packagemanager_aidl interface used to be part
of libbinder prior to Android 13:
https://android.googlesource.com/platform/frameworks/native/+/8cd952395060fa9215d3ba72113562f18d5f636a

Change-Id: I7232b9cd15b48d178e5af65a387562fa7df768d9

mlehtima commented on Android.mk:
```diff
@@ -251,11 +251,11 @@ ifeq ($(ANDROID_MAJOR),$(filter $(ANDROID_MAJOR),8 9))
 endif
 
-ifeq ($(ANDROID_MAJOR),$(filter $(ANDROID_MAJOR),10))
+ifeq ($(ANDROID_MAJOR),$(filter $(ANDROID_MAJOR),10 12))
```
Is this really needed and the change to the entry below is not enough?

NotKit replied:
Seems to be redundant for both, removed.
